code/go/0chain.net/core/ememorystore/connection.go

```diff
@@ -215,9 +218,12 @@ func Close(ctx context.Context) {
 		con.ReadOptions.Destroy()
 		con.WriteOptions.Destroy()
 		con.TransactionOptions.Destroy()
-		if err := con.Conn.Rollback(); err != nil {
-			logging.Logger.Warn("rollback failed", zap.Error(err))
-		} // commit is expected to be done by the caller of the get connection
+		if con.shouldRollback {
```

Comment:
is it safe to share connection between goroutines?

Reply:
@dabasov  Not sure yet. have to review all related code for it. It might be better to raise a new issue to check thread-safe feature on ememorystore.

Reply:
Instead of adding new potential race condition, I prefer to just leave the rollback warning message or please make sure the fix won't cause new problems.

Reply:
@peterlimg  In old code, transaction always be rollbacked whatever it is committed or not. that is why the warning always be thrown on logging. 
In new code, it is only be rollbacked when it is not committed. And `rollback` is default behavior. 
And it is only be skipped when `Commit` is invoked without any error. so I think it should not cause new problem.

Reply:
@dabasov  added `lock` for `Commit`